### PR TITLE
Ghost node no longer shown when hovering over existing node

### PR DESCRIPTION
**Pull request by arkadR**

### CODE REVIEW

File: Assets/Scripts/Tools/NodeTool.cs [hunk 1]
```diff
@@ -10,7 +10,7 @@ public class NodeTool : MonoBehaviour, ITool, IToolChangeObserver
     GraphService _graphService;
     ToolPanelController _toolPanelController;
     private Camera _attachedCamera;
-    private bool _shouldGhostNodeBeActive = false;
+    bool _lastCanInteractWithResult = false;
     private GameObject _ghostNode;
     private const float c_maxInteractDistance = 10f;
     private const float c_addNodeDistance = 4f;
```
Comment (efodiks) [inline Assets/Scripts/Tools/NodeTool.cs:16]: why this style for naming consts?

File: Assets/Scripts/Tools/EdgeTool.cs [hunk 2]
```diff
@@ -24,11 +24,12 @@ public void Start()
...
 
-    public bool CanInteractWith(RaycastHit hitInfo)
-    {
-      return _graphService.IsNode(hitInfo.collider.gameObject) 
-        || _graphService.IsEdge(hitInfo.collider.gameObject) && _previouslyHitNode == null;
-    }
+    public bool CanInteractWith(RaycastHit hitInfo) =>
+      _graphService.IsNode(hitInfo.collider.gameObject) || 
+      _graphService.IsEdge(hitInfo.collider.gameObject) && 
+      _previouslyHitNode == null;
```
Comment (efodiks) [inline Assets/Scripts/Tools/EdgeTool.cs:30]: I would like parentheses here to clarify precedence

File: Assets/Scripts/Tools/NodeTool.cs (hunk 1 above)
Comment (arkadR) [reply Assets/Scripts/Tools/NodeTool.cs:16]: I am pretty sure I mentioned this style when we were discussing it.
I am just used to it and have not found a nicer style for them. The default we have now is PascalCase but that's just confusing with properties. Const should have some indication that they are indeed consts so a normal private _ is not ideal as well. c_ may not look nice, but does the job and it's nice in the editor.

File: Assets/Scripts/Tools/NodeTool.cs [hunk 3]
```diff
@@ -20,26 +20,31 @@ public class NodeTool : MonoBehaviour, ITool, IToolChangeObserver
...
     }
 
     public void Update()
     {
-      if (!_shouldGhostNodeBeActive)
-        return;
-
-      var position = _attachedCamera.transform.position + (_attachedCamera.transform.forward * c_addNodeDistance);
-      _ghostNode.transform.position = position;
+      if (_movedNode != null || _lastCanInteractWithResult == true)
```
Comment (karol7531) [inline Assets/Scripts/Tools/NodeTool.cs:31]: nice, 100x better now

File: Assets/Scripts/Tools/NodeTool.cs [hunk 4]
```diff
@@ -95,24 +96,21 @@ public void OnRightClick(Transform cameraTransform, bool isRayCastHit, RaycastHi
 
     public void OnSelect()
     {
-      _shouldGhostNodeBeActive = true;
-      if (_toolPanelController != null)
-        _toolPanelController.SetBackgroundColor(Color.green);
-      if (_ghostNode != null)
-        _ghostNode.SetActive(true);
+      enabled = true;
```
Comment (karol7531) [inline Assets/Scripts/Tools/NodeTool.cs:99]: doesn't it throw NullPointerException?

Comment (arkadR) [reply Assets/Scripts/Tools/NodeTool.cs:99]: enabled = true?
Nope

**Pull request merged**